src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNet/SqlCommandIntegration.cs

```diff
@@ -48,13 +68,13 @@ public static class SqlCommandIntegration
 
             try
             {
-                var targetType = command.GetInstrumentedType(SqlCommandTypeName);
+                var targetType = command.GetInstrumentedType(SqlCommandTypeName, SqlClientNamespaces);
 
                 instrumentedMethod =
                     MethodBuilder<Func<object, object>>
                        .Start(moduleVersionPtr, mdToken, opCode, methodName)
                        .WithConcreteType(targetType)
-                       .WithNamespaceAndNameFilters(SqlDataReaderTypeName)
+                       .WithNamespaceAndNameFilters(SystemSqlDataReaderTypeName)
```

Comment:
This doesn't seem right. If this is instrumenting a call to Microsoft.Data.SqlClient, then the argument would be of type `MicrosoftSqlDataReaderTypeName` right?

Reply:
Hmm so this only seems to be used when we can't resolve the metadata token (likely for types with generic type parameters) so we never actually consume this, but I would feel safer duplicating the code versus getting this wrong

Reply:
Good catch.

Reply:
In c1d01a27, I split every wrapper into 2 one-liners so each one can pass different values for type names.